x-pack/plugins/elastic_assistant/server/ai_assistant_data_clients/defend_insights/helpers.ts

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { get as _get, isArray } from 'lodash';
+
+import type { estypes } from '@elastic/elasticsearch';
+import type { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/types';
+import type { AuthenticatedUser } from '@kbn/core/server';
+import type {
+  DefendInsightCreateProps,
+  DefendInsightUpdateProps,
+  DefendInsightsResponse,
+  DefendInsightsGetRequestQuery,
+} from '@kbn/elastic-assistant-common';
+
+import type {
+  CreateDefendInsightSchema,
+  EsDefendInsightSchema,
+  UpdateDefendInsightSchema,
+} from './types';
+
+export const transformESSearchToDefendInsights = (
+  response: estypes.SearchResponse<EsDefendInsightSchema>
+): DefendInsightsResponse[] => {
+  return response.hits.hits
+    .filter((hit) => hit._source !== undefined)
+    .map((hit) => {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const insightSchema = hit._source!;
+      const defendInsight: DefendInsightsResponse = {
+        timestamp: insightSchema['@timestamp'],
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        id: hit._id!,
+        backingIndex: hit._index,
+        createdAt: insightSchema.created_at,
+        updatedAt: insightSchema.updated_at,
+        lastViewedAt: insightSchema.last_viewed_at,
+        users:
```

Comment:
Honestly, I copied this over from attack discovery and just changed the necessary keys 😅. My impression though is that we want to ensure we don't potentially leak additional fields? and maybe also just because new objects.